cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterCreator.java

```diff
@@ -328,14 +332,20 @@ public static KafkaCluster createKafkaCluster(
             Map<String, Storage> oldStorage,
             Map<String, List<String>> currentPods,
             KafkaVersionChange versionChange,
-            boolean useKRaftEnabled,
+            KafkaMetadataStateManager kafkaMetadataStateManager,
             KafkaVersion.Lookup versions,
             SharedEnvironmentProvider sharedEnvironmentProvider
     ) {
-        boolean isKRaftEnabled = useKRaftEnabled && ReconcilerUtils.kraftEnabled(kafkaCr);
-        List<KafkaPool> pools = NodePoolUtils.createKafkaPools(reconciliation, kafkaCr, nodePoolCrs, oldStorage, currentPods, isKRaftEnabled, sharedEnvironmentProvider);
-        String clusterId = isKRaftEnabled ? NodePoolUtils.getOrGenerateKRaftClusterId(kafkaCr, nodePoolCrs) : NodePoolUtils.getClusterIdIfSet(kafkaCr, nodePoolCrs);
-        return KafkaCluster.fromCrd(reconciliation, kafkaCr, pools, versions, versionChange, isKRaftEnabled, clusterId, sharedEnvironmentProvider);
+        KafkaMetadataConfigurationState kafkaMetadataConfigState = kafkaMetadataStateManager.getMetadataConfigurationState();
+        LOGGER.infoCr(reconciliation, "KafkaMetadataConfigurationState = {}", kafkaMetadataConfigState);
```

Comment:
eft over log message

Reply:
Yes, I will review all INFO level logging when we are close to accept the PR. They are still useful to me for development/testing during the review as well.